repos/system_upgrade/el8toel9/actors/scanblacklistca/libraries/scanblacklistca.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import BlackListCA, BlackListError
+
+# dict(orig_dir: new_dir)
+DIRS_CHANGE = {
+    '/etc/pki/ca-trust/source/blacklist/': '/etc/pki/ca-trust/source/blocklist/',
+    '/usr/share/pki/ca-trust-source/blacklist/': '/usr/share/pki/ca-trust-source/blocklist/'
+}
+
+
+def _get_dirs():
+    return DIRS_CHANGE
+
+
+def _get_files(dirname):
+    # on rhel8, -type can't take two arguments, so we need to call find
+    # twice and concatenate the results
+    files = run(['find', dirname, '-type', 'f'], split=True)['stdout']
+    return files + run(['find', dirname, '-type', 'l'], split=True)['stdout']
+
+
+def _make_messages(dirname, targetname):
+    if not os.path.exists(dirname):
+        # The directory does not exist; not an error (there is just nothing
+        # to migrate).
+        return
+    try:
+        blacklisted_certs = _get_files(dirname)
+    except (CalledProcessError, OSError) as e:
```

Comment:
Just curious, when can we end up with OSError? iiuc _get_files raises CalledProcessError only in case the command didn't succeed https://github.com/oamg/leapp/blob/master/leapp/libraries/stdlib/__init__.py#L185

Reply:
I got it from 'migrateblacklistca.py'. it was probably here because maybe a version of __get_files had some os. calls at one point, or just paranoia. I can remove it if we know that run only throws CalledProcessError.